### PR TITLE
feat(EMS-669): Company Details - Broker - Postcode Validation

### PR DESCRIPTION
**Pull request by Zainzzkk**
# Changes made:
* Used npm package `postcode-validator` for postcode validation
* Added shared validation for postcode
* Added unit and e2e tests

### CODE REVIEW

File: e2e-tests/cypress/e2e/journeys/insurance/your-business/broker/validation/broker-postcode-validation.spec.js [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+    it('should display validation errors', () => {
+      const field = broker[FIELD_ID];
+
+      field.yesRadioInput().click();
+
+      const {
+        errorField, errorIndex, expectedErrorsCount, errorMessageFormat,
+      } = ERROR_ASSERTIONS;
+
+      const inputValue = 'SW1 22A';
```
Comment (ttbarnes) [inline e2e-tests/cypress/e2e/journeys/insurance/your-business/broker/validation/broker-postcode-validation.spec.js:121]: Could we move all the values into the `examples` directory like we have for phone and website? `phone-number-examples.js`

File: e2e-tests/cypress/e2e/journeys/insurance/your-business/broker/validation/broker-postcode-validation.spec.js [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+        errorField, errorIndex, expectedErrorsCount, errorMessageFormat,
+      } = ERROR_ASSERTIONS;
+
+      const inputValue = 'S 2AA';
+
+      cy.submitAndAssertFieldErrors(errorField, inputValue, errorIndex, expectedErrorsCount, errorMessageFormat);
+    });
+  });
+
+  describe('when the postcode has a space', () => {
```
Comment (ttbarnes) [inline e2e-tests/cypress/e2e/journeys/insurance/your-business/broker/validation/broker-postcode-validation.spec.js:191]: Can we wrap the last 2 tests in a `describe('valid postcode ...` so it's clearer?

File: src/ui/server/controllers/insurance/business/broker/validation/rules/broker-postcode.test.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    expect(response).toEqual(mockErrors);
+  });
+
+  it('should return the mockErrors if using broker is "null"', () => {
+    mockBody[USING_BROKER] = null;
+
+    const response = brokerPostcode(mockBody, mockErrors);
+
+    expect(response).toEqual(mockErrors);
+  });
```
Comment (ttbarnes) [inline src/ui/server/controllers/insurance/business/broker/validation/rules/broker-postcode.test.ts:48]: Can we wrap the yes/no/null parts into describe blocks please like `broker-email`, easier to read/digest 😊

File: e2e-tests/cypress/e2e/journeys/insurance/your-business/broker/validation/are-you-using-a-broker-answer-yes.spec.js [hunk 4]
```diff
@@ -93,6 +94,7 @@ context('Insurance - Your business - Broker Page - As an Exporter I want to conf
       broker[ADDRESS_LINE_1].input().clear().type('Address line 1');
       broker[TOWN].input().clear().type('town');
       broker[EMAIL].input().clear().type('test@test.com');
+      broker[POSTCODE].input().clear().type('SW1A 2AA');
```
[the remark this answers is not in the document]
Comment (Zainzzkk) [reply e2e-tests/cypress/e2e/journeys/insurance/your-business/broker/validation/are-you-using-a-broker-answer-yes.spec.js:97]: I will be doing this on save and go back

File: e2e-tests/constants/examples/postcode-examples.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
+export const INVALID_POSTCODES = {
+  ONE_LETTER_FIRST_PART: 'S 2AA',
+  NO_LETTERS_FIRST_PART: '22 2AA',
+  TWO_DIGITS_SECOND_PART: 'SW1 22A',
+  THREE_DIGITS_SECOND_PART: 'SW1 222',
+  ALL_LETTERS_SECOND_PART: 'SW1 AAA',
+  TOO_MANY_CHARACTERS: 'SW1A 2AAA',
+  TOO_FEW_CHARACTERS: 'S 2AA',
```
Comment (ttbarnes) [inline e2e-tests/constants/examples/postcode-examples.js:8]: For the too many & too few tests, should we have tests for with spaces and without spaces?

**Review by ttbarnes — APPROVED** [review]

**Pull request merged**